core/execution/future/market.go

```diff
@@ -761,7 +761,7 @@ func (m *Market) GetMarketData() types.MarketData {
 			if internalCompositePrice == nil {
 				internalCompositePrice = num.UintZero()
 			} else {
-				internalCompositePrice = m.priceToMarketPrecision(internalCompositePrice)
+				internalCompositePrice = m.priceToMarketPrecision(internalCompositePrice.Clone())
```

Comment:
Wouldn't it make more sense to have `CompositePriceCalculator.GetPrice()` return a cloned value? Relying on the caller to clone the value feels like we're not addressing the root cause

Reply:
Good point. That method doesn't exist in `v0.74.1` which I was debugging in, I'll update.